pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/parser/RuntimeASTTransformer.groovy

```diff
@@ -50,16 +57,94 @@ import javax.annotation.Nonnull
 
 import static org.codehaus.groovy.ast.tools.GeneralUtils.*
 import static org.jenkinsci.plugins.pipeline.modeldefinition.parser.ASTParserUtils.*
+import static org.objectweb.asm.Opcodes.ACC_FINAL
+import static org.objectweb.asm.Opcodes.ACC_PUBLIC
+import static org.objectweb.asm.Opcodes.ACC_STATIC
 
 /**
  * Transforms a given {@link ModelASTPipelineDef} into the {@link Root} used for actual runtime. Also attaches a
  * {@link ExecutionModelAction} to the run.
  */
 @SuppressFBWarnings(value="SE_NO_SERIALVERSIONID")
 class RuntimeASTTransformer {
-    RuntimeASTTransformer() {
+    ClassNode rootClassNode
+    ModuleNode moduleNode
+    def methodClassNode = [:]
+    def methodCount = [:]
+    def constLookup = [:]
+//    def variableClassNode = null
+//    def variableValues = [:]
+
+
+    RuntimeASTTransformer(ModuleNode moduleNode) {
+        this.moduleNode = moduleNode
+    }
+
+    Expression ctorXFunction(ClassNode type, Expression args) {
```

Comment:
Helper function that turns a ConstructorExpression into a MethodCall to a generated static method that returns that ConstructorExpression.

Reply:
This is some wonderfully demented magic. I approve.